src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1110,6 +1112,35 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                 );
                 return;
             }
+
+        }
+
+        if let &Place::Projection(_) = place {
+            if let Some(local) = place.base_local() {
+                let mpi = self.move_data.rev_lookup.find_local(local);
+                if flow_state.uninits.contains(mpi) {
+                    self.used_mut.insert(local);
```

Comment:
Comment: 

Pre-emptively insert local into the `used_mut` set to avoid any warnings related to whether the `mut` declaration is used.

Reply:
We may want to check whether `local` is actually *declared* as mut, something like

```rust
if let Mutability::Mut = self.mir.local_decls[local].mutability {
    self.used_mut.insert(local);
}
```